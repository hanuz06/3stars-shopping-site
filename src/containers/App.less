@import "~antd/dist/antd.less";

* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

html,
body {
  height: 100%;
}

ul,
li {
  display: block;
  padding: 0;
  margin: 0;
  list-style-type: none;
}

img {
  display: block;
}
