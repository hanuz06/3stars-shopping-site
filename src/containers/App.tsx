import React, { FC } from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
  Redirect,
} from "react-router-dom";
import "./App.less";
import LandingPage from "./LandingPage";
import LoginRegistrationPage from "./LoginRegistrationPage";
import CheckoutPage from "./CheckoutPage";

import { Layout } from "antd";
import HeaderContainer from "./HeaderContainer";
// import ProductDetailsPage from "./ProductDetailsPage";
// import CartPage from "./CartPage";
import CartItem from "./CartItem";
import CartList from "./CartList";
import CartPage from "./CartPage";

const { Header, Content, Footer } = Layout;

const App: FC = () => {
  return (
    <Router>
      <div className="App">
        <Layout style={{position:"relative"}}>
          {/* <Header> */}
            <HeaderContainer />
          {/* </Header> */}
          <Content>
            <Switch>
              <Route exact path="/" component={LandingPage} />
              <Route path="/login" component={LoginRegistrationPage} />
              <Route path="/cart" component={CartPage} />
              <Route path="/checkout" component={CheckoutPage} />
            </Switch>
          </Content>

          {/* <Footer
            style={{
              // position: "fixed",
              textAlign: "center",
              width: "100%",
              // left: "auto",
              // right: "auto",
              bottom: 0,
            }}
          >
            3stars Corp., 2020
          </Footer> */}
        </Layout>
      </div>
    </Router>
  );
};
export default App;
