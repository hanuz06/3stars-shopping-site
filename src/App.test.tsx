import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

import renderer from "react-test-renderer";

test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/Button/i);
  expect(linkElement).toBeInTheDocument();
});

describe("<App />", () => {
  it("match to snapshot", () => {
    const tree = renderer.create(<App />).toJSON();   
    expect(tree).toMatchSnapshot();  
  });
});