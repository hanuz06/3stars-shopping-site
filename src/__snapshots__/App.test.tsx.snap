// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> match to snapshot 1`] = `
<div
  className="App"
>
  <button
    className="ant-btn ant-btn-primary"
    onClick={[Function]}
    type="button"
  >
    <span>
      Button
    </span>
  </button>
</div>
`;
